app/benchmarking.py

```diff
@@ -56,54 +61,190 @@ def run(circuit, backend, token, shots, benchmark_id, original_depth, original_w
-                               benchmark_id=benchmark_id,
-                               original_depth=i, original_width=num_of_qubits,
-                               original_number_of_multi_qubit_gates=original_number_of_multi_qubit_gates,
-                               transpiled_depth=transpiled_depth_sim, transpiled_width=transpiled_width_sim,
-                               transpiled_number_of_multi_qubit_gates=transpiled_number_of_multi_qubit_gates_sim)
-
-            location_real = run(circuit=qcircuit_real, backend=qpu_name, token=token, shots=shots,
-                                benchmark_id=benchmark_id,
-                                original_depth=i, original_width=num_of_qubits,
-                                original_number_of_multi_qubit_gates=original_number_of_multi_qubit_gates,
-                                transpiled_depth=transpiled_depth_real, transpiled_width=transpiled_width_real,
-                                transpiled_number_of_multi_qubit_gates=transpiled_number_of_multi_qubit_gates_real)
-            location_benchmark = '/qiskit-service/api/v1.0/benchmarks/' + str(benchmark_id)
-            locations.append({'result-simulator': str(location_sim),
-                              'result-real-backend': str(location_real),
-                              'result-benchmark': str(location_benchmark)})
+    if clifford:
+        # create the given number of completely random clifford gate circuits of given width
+        provider = IBMQ.get_provider(group='open')
+        backend_real = provider.get_backend(qpu_name)
+        backend_sim = provider.get_backend(sim_name)
+        nseeds = num_of_circuits
+
+        # random rb_pattern with random number of multi-qubit-gates
+        qubit_seq = [j for j in range(num_of_qubits)]
+        rb_pattern = []
+        random.shuffle(qubit_seq)
+        while qubit_seq:
+            n = random.randint(1, len(qubit_seq))
+            rb_pattern.append(qubit_seq[0:n])
+            del qubit_seq[0:n]
+
+        nCliffs = [random.randint(min_depth_of_circuit, max_depth_of_circuit) for _ in range(num_of_circuits)]
+        rb_opts = {}
+        rb_opts['length_vector'] = nCliffs
+        rb_opts['nseeds'] = nseeds
+        rb_opts['rb_pattern'] = rb_pattern
+
+        rb_circs, xdata = rb.randomized_benchmarking_seq(**rb_opts)
+        count = 0
+        index_list_elem = 0
+
+        for listElem in rb_circs:
+            index_list_elem += 1
+            count += len(listElem)
+            for indexElem, elem in enumerate(listElem):
+                rowcount = db.session.query(Benchmark).count()
+                benchmark_id = rowcount // 2
+
+                remove_final_meas = RemoveFinalMeasurements()
+
+                transpiled_circuit_real = transpile(elem, backend=backend_real, optimization_level=3)
+                transpiled_circuit_sim = transpile(elem, backend=backend_sim, optimization_level=3)
+
+                active_qubits_transpiled_circuit_real = [
```

Comment:
from here til the `else`, isn't it duplicate code compared with the `else` code? If so write it out of the loop